api/src/main/java/uk/gov/hmcts/bar/api/data/repository/PaymentInstructionsSpecifications.java

```diff
@@ -226,13 +226,14 @@ public Predicate toPredicate(Root<PaymentInstruction> root, CriteriaQuery<?> que
 
         @Override
         public Predicate toPredicate(Root<PaymentInstruction> root, CriteriaQuery<?> query, CriteriaBuilder criteriaBuilder) {
-            Predicate predicate = null;
+            In<PaymentType> inCriteriaForPaymentType  = null;
             Join<PaymentInstruction, PaymentType> paymentType = root.join("paymentType");
             if (paymentInstructionSearchCriteriaDto.getPaymentType() != null) {
-                predicate = criteriaBuilder.equal(paymentType.get("id"), paymentInstructionSearchCriteriaDto.getPaymentType());
+                inCriteriaForPaymentType = criteriaBuilder.in(paymentType);
             }
-            return predicate;
+            return Util.getListOfPaymentTypes(inCriteriaForPaymentType, paymentInstructionSearchCriteriaDto.getPaymentType());
```

Comment:
This can be written like:

if (paymentInstructionSearchCriteriaDto.getPaymentType() != null) {
                inCriteriaForPaymentType = criteriaBuilder.in(paymentType);
                inCriteriaForPaymentType = Util.getListOfPaymentTypes(inCriteriaForPaymentType, paymentInstructionSearchCriteriaDto.getPaymentType());
            }
            return inCriteriaForPaymentType;